src/compiler/emitter.ts

```diff
@@ -4434,19 +4458,14 @@ module ts {
                         writeLine();
                         emitLeadingComments(member);
                         emitStart(member);
-                        emitStart((<MethodDeclaration>member).name);
                         emitDeclarationName(node);
                         if (!(member.flags & NodeFlags.Static)) {
                             write(".prototype");
                         }
                         emitMemberAccessForPropertyName((<MethodDeclaration>member).name);
-                        emitEnd((<MethodDeclaration>member).name);
                         write(" = ");
-                        // TODO (drosen): Should we performing emitStart twice on emitStart(member)?
-                        emitStart(member);
                         emitFunctionDeclaration(<MethodDeclaration>member);
                         emitEnd(member);
-                        emitEnd(member);

```

Comment:
Actually, it looks correct. The one thing I am not sure about, is whether we should emitStart/End on member or member.name